hack/generator/pkg/config/type_transformer.go

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package config
+
+import (
+	"fmt"
+	"github.com/Azure/k8s-infra/hack/generator/pkg/astmodel"
+	"regexp"
+)
+
+type TransformTarget struct {
+	PackagePath string `yaml:",omitempty"`
+	Name        string `yaml:",omitempty"`
+}
```

Comment:
Do these satisfy the "Is-A" guideline for when to embed vs name?

Reply:
I'm not sure why `TransformTarget` needs to be a nested struct, why not have optional fields directly in `TypeTransformer` to capture the changes?

Reply:
The reason I did it this way is so that the transformer was clearly:
```
// Properties to select types to apply to (what I am now calling TypeMatcher)

// A single property describing the type to transform to
```

I'm not really married to it though, we could do away with this entirely and promote it up a level